Content/ContentResolver.php

```diff
@@ -23,12 +24,9 @@ class ContentResolver implements ContentResolverInterface
      */
     private $resolvers;
 
-    /**
-     * @param ContentTypeResolverInterface[] $resolvers
-     */
-    public function __construct(array $resolvers)
+    public function __construct(RewindableGenerator $resolvers)
     {
-        $this->resolvers = $resolvers;
+        $this->resolvers = iterator_to_array($resolvers);
```

Comment:
is this really needed? i would love to see that we does not need this - else we wont have to resolve the services if not needed

Reply:
@wachterjohannes the services will not resolved aslong as we call not a function on them. So iterator_to_array will not have a effect and is at current state the only way to convert the generator to keys and is the [documented](https://symfony.com/blog/new-in-symfony-4-3-indexed-and-tagged-service-collections) way by symfony.

Reply:
OK then everything is good